worlds/sc2/__init__.py

```diff
@@ -136,293 +174,386 @@ def setup_events(player: int, locked_locations: typing.List[str], location_cache
+                if item.name not in nco_support_items
+                or nco_support_items[item.name] in possible_starter_items
+                and ((ItemFilterFlags.Plando|ItemFilterFlags.Excluded) & possible_starter_items[nco_support_items[item.name]].flags) == 0
+            ]
+        if not basic_unit_options:
+            raise Exception("Early Unit: At least one basic unit must be included")
+        local_basic_unit = [item for item in basic_unit_options if ItemFilterFlags.NonLocal not in item.flags]
+        if local_basic_unit:
+            basic_unit_options = local_basic_unit
+        
+        unit = world.random.choice(basic_unit_options)
+        unit.flags |= ItemFilterFlags.StartInventory
+
+        # NCO-only specific rules
+        if first_mission == SC2Mission.SUDDEN_STRIKE:
+            if unit.name in nco_support_items:
+                support_item = possible_starter_items[nco_support_items[unit.name]]
+                support_item.flags |= ItemFilterFlags.StartInventory
+            if ItemNames.NOVA_JUMP_SUIT_MODULE in possible_starter_items:
+                possible_starter_items[ItemNames.NOVA_JUMP_SUIT_MODULE].flags |= ItemFilterFlags.StartInventory
+        if MissionFlag.Nova in first_mission.flags:
+            possible_starter_weapons = (
+                ItemNames.NOVA_HELLFIRE_SHOTGUN,
+                ItemNames.NOVA_PLASMA_RIFLE,
+                ItemNames.NOVA_PULSE_GRENADES,
+            )
+            starter_weapon_options = [item for item in possible_starter_items.values() if item.name in possible_starter_weapons]
+            starter_weapon = world.random.choice(starter_weapon_options)
+            starter_weapon.flags |= ItemFilterFlags.StartInventory
+        mission_count = sum(len(campaign) for campaign in world.mission_req_table.values())
+        if mission_count <= 10 and world.final_mission_id == SC2Mission.END_GAME.id:
+            if ItemNames.LIBERATOR_RAID_ARTILLERY in possible_starter_items:
+                possible_starter_items[ItemNames.LIBERATOR_RAID_ARTILLERY].flags |= ItemFilterFlags.StartInventory
             else:
-                pool.append(item)
+                logger.warning(f"Tried and failed to add {ItemNames.LIBERATOR_RAID_ARTILLERY} to the start inventory because it was excluded or plando'd")
 
-    existing_items = starter_items + [item for item in world.multiworld.precollected_items[world.player] if item not in starter_items]
-    existing_names = [item.name for item in existing_items]
 
-    # Check the parent item integrity, exclude items
-    pool[:] = [item for item in pool if pool_contains_parent(item, pool + locked_items + existing_items)]
+def flag_start_abilities(world: SC2World, item_list: List[FilterItem]) -> None:
+    starter_abilities = world.options.start_primary_abilities
+    if not starter_abilities:
+        return
+    assert starter_abilities <= 4
+    ability_count = int(starter_abilities)
+    ability_tiers = [0, 1, 3]
```

Comment:
Instead of ability tiers I'd just use distinction between ultimate and non-ultimate ones as coop abilities may arrive later. It could be a subject for another PR

Reply:
Yeah, this code is functionally unchanged from the old stuff, git has just decided to put the old code distant in the PR (it's old line 283 or so). HotS is the campaign I play and enjoy the least, so I'm probably not the one to make decisions about it. I agree that's a task for another PR.